atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2015 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs.impl.oracle;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.dbkvs.DbKeyValueServiceConfiguration;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.ConnectionSupplier;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.DbDdlTable;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.OverflowMigrationState;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.TableSize;
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.exception.PalantirSqlException;
+import com.palantir.nexus.db.sql.AgnosticResultSet;
+import com.palantir.util.VersionStrings;
+
+public class OracleDdlTable implements DbDdlTable {
+    private static final Logger log = LoggerFactory.getLogger(OracleDdlTable.class);
+    private final String tableName;
+    private final ConnectionSupplier conns;
+    private final OverflowMigrationState migrationState;
+    private final DbKeyValueServiceConfiguration config;
+
+    public OracleDdlTable(String tableName,
+                          ConnectionSupplier conns,
+                          OverflowMigrationState migrationState,
+                          DbKeyValueServiceConfiguration config) {
+        this.tableName = tableName;
+        this.conns = conns;
+        this.migrationState = migrationState;
+        this.config = config;
+    }
+
+    @Override
+    public void create(byte[] tableMetadata) {
+        if (conns.get().selectExistsUnregisteredQuery(
+                "SELECT 1 FROM pt_metropolis_table_meta WHERE table_name = ?",
```

Comment:
if we want to change away from pt_metropolis, we should do it now and put an upgrade task in our side, since most oss clients currently don't have upgrade tasks.

We can also get rid of pt_met if we pass in a setting for a table prefix; that way we can keep pt_met_ out of the oss code entirely and the average oss user of atlas will just neglect to supply a table prefix and just have tables named the proper actual tablename